### PR TITLE
Muti- DimensionalArrays API

### PR DESCRIPTION
**Pull request by raksha009**
*References to other Issues or PRs or Relevant literature*
*Refrenced Issue* #104 

Brief description of what is fixed or changed

Proposal of an API of Multi-Dimensional Arrays

Other comments
This is how I plan to implement this API and the role of each method and data- member

### CODE REVIEW

File: pydatastructs/linear_data_structures/Mutidimensional Array.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+Methods
+1. __new__(cls, dtype=NoneType, *args, **kwargs)
+2. __getitem__(self, *indices)
+3. __setitem__(self, value, *indices)
+4. fill(self, elem)
+
+'''
+
+class MutiDimensionalArray(object):
```
Comment (manaswinidas) [inline pydatastructs/linear_data_structures/Mutidimensional Array.py]: please make sure there is no typo and take care of the failing tests too

Comment (raksha009) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: Its just proposed API, i was asked to propose an API first before the implementation part... Plz review and suggest possible changes in my proposal.
@manaswinidas @czgdp1807

Comment (manaswinidas) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: I guess APIs are proposed in the issue itself

Comment (raksha009) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: Idk, i dm @czgdp1807 sir,  we decided that i would propose API in comments of a PR but I wasn't able to create a PR just for comments, so I described the working of methods in attached python file. Plz guide me about the workflow, i am quite confused now.

File: pydatastructs/linear_data_structures/Mutidimensional Array.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+'''
+Data Members
+1. _ndim - The number of dimensions in the array.
+2. _data - The python list which will store the data in row major order.
+3. _dtype - The data type of the elements.
+
+Methods
+1. __new__(cls, dtype=NoneType, *args, **kwargs)
+2. __getitem__(self, *indices)
+3. __setitem__(self, value, *indices)
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/Mutidimensional Array.py]: Can we have `__setitem__(self, *indices_value)`? `value` can be obtained from `value = indices_value[-1]`.

File: pydatastructs/linear_data_structures/Mutidimensional Array.py (hunk 1 above)
Comment (czgdp1807) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: The PR is fine, but the code for multidimensional array should be appended to [this file](https://github.com/codezonediitj/pydatastructs/blob/master/pydatastructs/linear_data_structures/arrays.py).

Comment (czgdp1807) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: `MultiDimensionalArray` should inherit abstract [Array](https://github.com/codezonediitj/pydatastructs/blob/b0c3136dfaa6d8551adfa208bcdc76bb3f95dd4f/pydatastructs/linear_data_structures/arrays.py#L8) class.

File: pydatastructs/linear_data_structures/Mutidimensional Array.py (hunk 2 above)
Comment (raksha009) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: Yes, okay

File: pydatastructs/linear_data_structures/Mutidimensional Array.py (hunk 1 above)
Comment (raksha009) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: Okk, thanks. I'll start working on the implementation part

**Comment by czgdp1807** [comment]
The API is fine. You can start with the implementation in this PR itself. Just push the changes/commits to your `patch1` branch.

File: pydatastructs/linear_data_structures/Mutidimensional Array.py (hunk 1 above)
Comment (czgdp1807) [reply pydatastructs/linear_data_structures/Mutidimensional Array.py]: > I guess APIs are proposed in the issue itself

Yes, that should have been done but @raksha009 has done that here which is okay. Let's keep this open and push the changes here for the implementation.

**Comment by raksha009** [comment]
Okk👍

**Comment by czgdp1807** [comment]
@raksha009 Are you working on it?

**Comment by raksha009** [comment]
Hii, sorry for the delay. I was trying to read about various types of classes in python more deeply. But i am facing great difficulty in implementing the same.

**Comment by czgdp1807** [comment]
Can you please describe your problems?

**Comment by raksha009** [comment]
Problems in parameters like *args, *indices_value and how to extract info from them... I have never came across such "*" parameters. I tried searching online but still couldn't understand them that clearly.

**Comment by czgdp1807** [comment]
Try going through https://www.digitalocean.com/community/tutorials/how-to-use-args-and-kwargs-in-python-3

**Comment by raksha009** [comment]
Okk, thanks sir

**Comment by raksha009** [comment]
I made some changes. Plz review

**Comment by czgdp1807** [comment]
We are working on arbitrary dimensional arrays. The code you have written will work only for 2D matrices. Please modify it to support queries like, `q[i1][i2][i3]....[in]`.

**Comment by czgdp1807** [comment]
Any updates?

**Comment by raksha009** [comment]
Not right now, am still working on it.

**Comment by czgdp1807** [comment]
Closing in favour of #256

**Pull request closed without merging**